app/src/main/java/org/wikipedia/page/SidePanelHandler.kt

```diff
@@ -87,8 +108,80 @@ class ToCHandler internal constructor(private val fragment: PageFragment,
                     onStartShow()
                 }
             }
+
+            override fun onDrawerClosed(drawerView: View) {
+                super.onDrawerClosed(drawerView)
+                enableToCContainer(true)
+            }
         })
         setScrollerPosition()
+        enableToCContainer()
+    }
+
+    fun setupTalkTopics(pageTitle: PageTitle) {
+        val newPageTitle = pageTitle.copy()
+
+        talkProgressBar.isVisible = true
+        talkErrorView.visibility = View.GONE
+        talkEmptyContainer.visibility = View.GONE
+
+        talkRecyclerView.layoutManager = LinearLayoutManager(fragment.requireContext())
+        talkRecyclerView.addItemDecoration(FooterMarginItemDecoration(0, 120))
+        talkRecyclerView.addItemDecoration(DrawableItemDecoration(fragment.requireContext(), R.attr.side_panel_list_separator_drawable, drawStart = true, drawEnd = false))
+        talkRecyclerView.adapter = talkTopicsAdapter
+        talkErrorView.backClickListener = View.OnClickListener {
+            hide()
+        }
+
+        TalkTopicsProvider(newPageTitle).load(object : TalkTopicsProvider.Callback {
```

Comment:
If the `TalkTopicsProvider` takes a long time to complete the network requests, and the Fragment is destroyed, these callbacks will cause crashes.
We should somehow "clean up" by calling the `TalkTopicsProvider.cancel()` function when the Fragment is destroyed.

Reply:
Makes sense and done.